### PR TITLE
Docker tweaks

### PR DESCRIPTION
**Pull request by mostlygeek**
Found some bugs/tweaks in the Docker scripts that needed addressing. 

- add conf/run.sh to use as the ENTRYPOINT for the container
- added `exec` instruction to shell scripts, nicer for handling signals (SIGKIL, SIGINT, etc)
- fix tests, they actually run now and don't error out on test data

@chuckharmston r? I think this is ready to merge.

### CODE REVIEW

**Comment by chuckharmston** [comment]
We want builds to fail on linting errors. We prevent merges unless builds succeed, so it should never be an issue on the ops side.

File: circle.yml [hunk 1]
```diff
@@ -29,8 +29,7 @@ dependencies:
 # Run flake8 and the Python test suite via nose.
 test:
   override:
-    - docker run -d app:build sh -c 'flake8 . --exclude=app/src && nosetests --exclude-dir=app/src'
-
+    - docker run app:build test
```
Comment (chuckharmston) [inline circle.yml]: How does this work? Is it effectively `$ENTRYPOINT test`?

File: conf/run.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        exec ./web.sh
+        ;;
+    worker)
+        echo "Starting Celery Worker"
+        exec ./worker.sh
+        ;;
+    test)
+        echo "Running Tests"
+        cd ..
+        flake8 . --exclude=./recommendation/views/data/dummy.py && nosetests
```
Comment (chuckharmston) [inline conf/run.sh]: Why are we excluding the dummy data view?

File: Dockerfile [hunk 3]
```diff
@@ -8,15 +8,22 @@ MAINTAINER https://mail.mozilla.org/listinfo/testpilot-dev
...
+    apt-get -qq install -y libmemcached-dev=1.0.18-4
 
 COPY ./requirements.txt /app/requirements.txt
 RUN pip install --upgrade --no-cache-dir -r /app/requirements.txt
 
-COPY . /app
 ENV PYTHONPATH $PYTHONPATH:/app
+EXPOSE 8000
+
+# default to running the web server
```
Comment (chuckharmston) [inline Dockerfile]: Nit: we try to make comments be full sentences with capitals and punctuation and all that.

File: conf/run.sh (hunk 2 above)
Comment (chuckharmston) [inline conf/run.sh]: Oh, oops, that's the data. It should still pass linting, though.

**Comment by chuckharmston** [comment]
Would you mind squashing the commits, as well?

**Comment by chuckharmston** [comment]
Generally looks good! A few questions and a few nits above.

File: circle.yml (hunk 1 above)
Comment (mostlygeek) [inline circle.yml]: yes. The `CMD` instruction is overridden by arguments on the command line. The `ENTRYPOINT` is only overridden if you use the `--entrypoint` with `docker run`

File: conf/run.sh (hunk 2 above)
Comment (mostlygeek) [inline conf/run.sh]: it fails linting pretty badly. See the [circle build log #28](https://circleci.com/gh/mozilla/universal-search-recommendation/28) ... 

```
./recommendation/views/data/dummy.py:10:80: E501 line too long (92 > 79 characters)
./recommendation/views/data/dummy.py:16:80: E501 line too long (98 > 79 characters)
./recommendation/views/data/dummy.py:28:80: E501 line too long (114 > 79 characters)
./recommendation/views/data/dummy.py:34:80: E501 line too long (139 > 79 characters)
./recommendation/views/data/dummy.py:46:80: E501 line too long (89 > 79 characters)
./recommendation/views/data/dummy.py:52:80: E501 line too long (127 > 79 characters)
./recommendation/views/data/dummy.py:70:80: E501 line too long (106 > 79 characters)
./recommendation/views/data/dummy.py:82:80: E501 line too long (105 > 79 characters)
....
```

I'll leave it to you to decide the best way to make this file pass linting. Most of the violations appear to be an a large object definition and long URI values.

**Comment by mostlygeek** [comment]
When you R+ this, I'll squash all the commits into a single one. I don't know how to do that without destroying all the comment history in the PR. :)

**Comment by chuckharmston** [comment]
Ah, it's failing because linting wasn't being run in the container, and I left an unchanged file in the branch where I added it before merging. I'll follow up and fix that.

As long as comments are made on the pull request (i.e. either in the "Conversaion" or "Files Changed" tabs) and not on specific commits, they're maintained through squashes.

 Ship it! :shipit:

**Comment by mostlygeek** [comment]
@chuckharmston ok. commits squashed, all tests pass. Please merge this when you're ready and pave the way for #71.

**Comment by chuckharmston** [comment]
Thanks @mostlygeek!

**Pull request merged**